src/js/tracker.js

```diff
@@ -2052,12 +2034,34 @@
 		apiMethods.enableActivityTracking = function (minimumVisitLength, heartBeatDelay) {
 			if (minimumVisitLength === parseInt(minimumVisitLength, 10) &&
 				heartBeatDelay === parseInt(heartBeatDelay, 10)) {
-				activityTrackingEnabled = true;
-				configMinimumVisitTime = new Date().getTime() + minimumVisitLength * 1000;
-				configHeartBeatTimer = heartBeatDelay * 1000;
+				activityTrackingConfig = {
+					activityTrackingEnabled: true,
+					configMinimumVisitLength: minimumVisitLength,
+					configHeartBeatTimer: heartBeatDelay * 1000
+				}
+			} else {
+				helpers.warn(activityTrackingWarning)
+			}
+		};
+
+		/**
+		 * Enables page activity tracking (replaces collector ping with callback).
+		 *
+		 * @param int minimumVisitLength Seconds to wait before sending first page ping
+		 * @param int heartBeatDelay Seconds to wait between pings
+		 * @param function callback function called with ping data
```

Comment:
This is a good idea, particularly as it allows the page ping to be tied back to the page view which started the activity tracking.
I've passed this back on the object into the callback now.
i.e. 

```
            window.snowplow('enableActivityTrackingCallback', 10, 10, function (a) {
                console.log(a.context);
            }
```

As it is already an array, it can be passed in as the last parameter onto any 'track' calls that may be done in the activity tracking callback. I like the extra flexibility that this offers users.